src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -306,6 +304,7 @@ public CSharpOperationFactory(SemanticModel semanticModel)
                 case BoundKind.StackAllocArrayCreation:
                 case BoundKind.TypeExpression:
                 case BoundKind.TypeOrValueExpression:
+                case BoundKind.UnconvertedCollectionLiteralExpression:
```

Comment:
Please track removing this with an issue. We should not be seeing these expressions here, and if we are things are surviving initial binding that should not.